app/views/auctions/_list_item.html.erb

```diff
@@ -2,7 +2,7 @@
   <div class="usa-grid">
     <div class='usa-width-one-whole issue-label'>
       <h3 class='issue-title'>
-        <span class="usa-label-big <%= auction_label_class(auction) %>"><%= auction_label(auction) %></span> <a href="<%= auction_path(auction) %>"><%= auction.title %></a>
+        <span class="usa-label-big <%= auction.label_class %>"><%= auction.label %></span> <a href="<%= auction_path(auction) %>"><%= auction.title %></a>
```

Comment:
if statements are consistent places of errors, but also complexity. Small methods are easier to work with not just because there is less to absorb, but because we give the smaller concept a name (and hopefully a good name). Knowing that partial based on some named criterial will always be rendered is more helpful that trying to navigate an if-else branch in a view. There shouldn't be logic in views. I used to use Mustache on the server to that kind of logic.

In terms of where things belong ideally, I try to refrain from the ideal and just get really practical. When a class gets too long, I start to feel like I can't absorb what it is doing, and it must have many responsibilities. Then I try to figure out what those are. If those responsibilities are also somewhere else in the application then yanking them up to a shared space might make sense. Or moving them into a really small class that can be used in both places.

So long as things are small and well-named and not overly abstract, moving things around and changing abstractions is easy. So, I try not to think abstractly where something should be, but how to make things smaller and well named. Then it gets a lot easier to figure out what things belong together and how to get them to a better place.